aii-ks/src/main/perl/ks.pm

```diff
@@ -1587,6 +1601,40 @@ EOF
         ksfix_grub;
     }
 
+    # restore UEFI pxeboot first
+    if ($tree->{pxeboot}) {
+        print <<EOF;
+#
+# restore pxeboot as first boot order in case efibootmgr is around
+#
+efibootmgr=/usr/sbin/efibootmgr
+function join { local IFS=","; echo "\$*"; }
+if [ -x \$efibootmgr ]; then
+    ordertxt=\$(\$efibootmgr |grep BootOrder | sed "s/.*: \\?//")
+    order=(\$(echo \$ordertxt | tr ',' ' '))
+    pxe=(\$(\$efibootmgr -v | grep -ie ' \\(NIC\\|PXE\\|Network\\) ' | sed "s/*\\? .*//;s/^Boot//" | sort ))
```

Comment:
For some reason I'm finding this Bash-in-Perl hard to follow. What are you trying to match here?

I haven't got many UEFI systems lying around, but `efibootmgr -v | grep -ie ' \(NIC\|PXE\|Network\) '` doesn't seem to match anything on them, whereas removing the spaces does match something, but...
```
# efibootmgr -v | grep -ie '\(NIC\|PXE\|Network\)'
Boot0007* Onboard NIC	BBS(Network,Onboard NIC,0x0)AMBO
Boot000A  Onboard NIC(IPV4)	PciRoot(0x0)/Pci(0x19,0x0)/MAC(aabbccddeeff,0)/IPv4(0.0.0.0:0<->0.0.0.0:0,0,0)AMBO
Boot000B  Onboard NIC(IPV6)	PciRoot(0x0)/Pci(0x19,0x0)/MAC(aabbccddeeff,0)/IPv6([::]:<->[::]:,0,0)AMBO
```

Reply:
FYI that example was with efibootmgr version 0.12.

Reply:
Can't you simple take the current boot device from `efibootmgr` and set that as the first device in boot order?

At least you will keep the situation in whatever state it was.